apps/web/src/components/Shared/Footer/Locale.tsx

```diff
@@ -28,41 +29,80 @@ const Locale: FC = () => {
     location.reload();
   }, []);
 
+  const [popperTrigger, popperContainer] = usePopper({
+    placement: 'bottom-start',
+    modifiers: [
+      {
+        name: 'toggle-placement',
+        enabled: true,
+        phase: 'main',
+        fn: ({ state, instance }) => {
+          const popperElement = state?.elements?.popper;
+          const popperHeight = popperElement.clientHeight || 105;
+          const referenceElement = state?.elements?.reference;
+          const referenceElementDistance =
+            referenceElement?.getBoundingClientRect()?.bottom + popperHeight;
+          const windowHeight = window.innerHeight;
+          const togglePlacement = referenceElementDistance - windowHeight > 0;
+          if (togglePlacement) {
+            if (state.placement === 'bottom-start') {
+              instance.setOptions({
+                placement: 'top-start'
+              });
+            }
+          } else {
+            if (state.placement === 'top-start') {
+              instance.setOptions({
+                placement: 'bottom-start'
+              });
+            }
+          }
+        }
+      },
+      {
+        name: 'preventOverflow'
+      }
+    ]
+  });
```

Comment:
we have 10+ menus in the app, we need to make it reusable and small

Reply:
but this hook did not handle the situation when we are using with portal or teleport then floating element will not acnhor to refrence element example when scrolling . can you also add this.